packages/modelviewer.dev/examples/staging-and-camera-control.html

```diff
@@ -299,7 +299,52 @@ <h4>Event listeners can cooperate with camera-controls</h4>
       </div>
     </div>
   </div>
-
+  <div class="sample">
+    <div id="demo-container-8" class="demo"></div>
+    <div class="content">
+      <div class="wrapper">
+        <div class="heading">
+          <h2 class="demo-title">Add other interactions like rotating environment map</h2>
+          <h4>Use parameters with rest turntable to rotate only the environment light</h4>
+          <p>By using shift key plus left click mouse or touch pad drag, one can rotate
+			  the environment map to see how the environment background would cast light
+			  onto the object from different angles when it is rotated. Updating orbit to 
+			  new theta value and then reseting the theta value to stop turntable from 
+		  rotating. </p>
+        </div>
+        <example-snippet stamp-to="demo-container-8" highlight-as="html">
+          <template>
+			  <model-viewer id="pan-demo" camera-controls src="../shared-assets/models/Astronaut.glb" environment-image="../assets/whipple_creek_regional_park_04_1k.hdr" 				alt="A 3D model of an astronaut"></model-viewer>
+			  <script>
+				  const modelViewer = document.querySelector("model-viewer");
+	  
+				  const updateOrbit = () => {
+				    const orbit = modelViewer.getCameraOrbit();
+				    modelViewer.cameraOrbit = `${orbit.theta + 0.01}rad ${orbit.phi}rad ${orbit.radius}m`;
+				    modelViewer.jumpCameraToGoal();
+					requestAnimationFrame(() => updateOrbit());
+			  	    var isMouseDown = false;
+			  	    document.onmousedown = function() { isMouseDown = true  };
+			  	    document.onmouseup   = function() { isMouseDown = false };
+			  	    document.onmousemove = function(e) { 
+			  		  if(isMouseDown) { 
+			  	  	    if (e.shiftKey && e.which == 1) {
+
+			  				modelViewer.resetTurntableRotation(-1 * orbit.theta);
```

Comment:
Have you tried serving this example locally (`npm run serve`) and testing it? I'm curious if this has the desired effect when you first rotate and then shift-rotate?

Reply:
nope, I wasnt sure at exactly what stage this gets tested. But I can do that locally as well and let you know how it goes. Previously I tested with older model viewer code and that had no effect. But I believe I was not using the right version of model viewer at that time as resetTurnTable() didn't take any parameters.

Reply:
So I made all the above changes and I realised I had to npm install first and when I ran npm run build. I am getting bunch of error. Not sure about them. 

@google/3dom: ../../node_modules/@types/jest/index.d.ts(45,15): error TS2451: Cannot redeclare block-scoped variable 'expect'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2514,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'beforeEach' must be of type 'Lifecycle', but here has type 'HookFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2532,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'afterEach' must be of type 'Lifecycle', but here has type 'HookFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2548,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'describe' must be of type 'Describe', but here has type 'SuiteFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2569,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'xdescribe' must be of type 'Describe', but here has type 'PendingSuiteFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2583,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'it' must be of type 'It', but here has type 'TestFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2597,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'test' must be of type 'It', but here has type 'TestFunction'.
@google/3dom: ../../node_modules/@types/mocha/index.d.ts(2604,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'xit' must be of type 'It', but here has type 'PendingTestFunction'.
@google/3dom: src/end-to-end-spec.ts(16,20): error TS2307: Cannot find module 'three/examples/jsm/loaders/GLTFLoader' or its corresponding type declarations.
@google/3dom: src/end-to-end-spec.ts(17,24): error TS2307: Cannot find module 'three/src/core/Object3D.js' or its corresponding type declarations.
@google/3dom: src/end-to-end-spec.ts(18,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/end-to-end-spec.ts(19,20): error TS2307: Cannot find module 'three/src/objects/Mesh.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph-spec.ts(16,59): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph-spec.ts(17,35): error TS2307: Cannot find module 'three/examples/jsm/loaders/GLTFLoader.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph-spec.ts(18,29): error TS2307: Cannot find module 'three/examples/jsm/utils/SkeletonUtils.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(1,56): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(2,48): error TS2307: Cannot find module 'three/examples/jsm/loaders/GLTFLoader.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(70,27): error TS7006: Parameter 'gltfElementReference' implicitly has an 'any' type.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(70,49): error TS7006: Parameter 'threeObject' implicitly has an 'any' type.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(82,28): error TS7053: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'GLTF'.
@google/3dom: src/facade/three-js/correlated-scene-graph.ts(129,36): error TS7053: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'GLTF'.
@google/3dom: src/facade/three-js/image.ts(16,52): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/material-spec.ts(16,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/material-spec.ts(17,21): error TS2307: Cannot find module 'three/src/math/Color.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/material.ts(16,61): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-graft-spec.ts(16,24): error TS2307: Cannot find module 'three/src/core/Object3D.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-spec.ts(16,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-spec.ts(17,20): error TS2307: Cannot find module 'three/src/objects/Mesh.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/model-spec.ts(32,33): error TS7006: Parameter 'object' implicitly has an 'any' type.
@google/3dom: src/facade/three-js/model.ts(16,36): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/pbr-metallic-roughness-spec.ts(16,36): error TS2307: Cannot find module 'three/src/materials/MeshStandardMaterial.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/pbr-metallic-roughness-spec.ts(17,21): error TS2307: Cannot find module 'three/src/math/Color.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/pbr-metallic-roughness.ts(16,61): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/sampler-spec.ts(16,23): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/sampler.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/texture-info.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/texture-spec.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/texture.ts(16,39): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/three-dom-element-spec.ts(17,24): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/facade/three-js/three-dom-element-spec.ts(18,24): error TS2307: Cannot find module 'three/src/core/Object3D.js' or its corresponding type declarations.
@google/3dom: src/facade/three-js/three-dom-element.ts(16,43): error TS2307: Cannot find module 'three' or its corresponding type declarations.
@google/3dom: src/test-helpers.ts(16,38): error TS2307: Cannot find module 'three' or its corresponding type declarations.

here is a log.

Reply:
Did you follow the README? It looks like you may have missed `npm run bootstrap`? Also, what OS are you on?

Reply:
Oh snaps, you are right, running that command, will let you know. I am on macOS catalina.